snooty/parser.py

```diff
@@ -613,6 +621,29 @@ def __init__(
         self.asset_dg: "networkx.DiGraph[FileId]" = networkx.DiGraph()
         self._expensive_operation_cache = Cache()
 
+        published_branches: Optional[PublishedBranches] = self.get_parsed_branches()
```

Comment:
If this raises an exception should we report a diagnostic instead of crashing? (See line 589. Also we should report whatever we found on line 608)